bundles/binding/org.openhab.binding.openenergymonitor/src/main/java/org/openhab/binding/openenergymonitor/protocol/OpenEnergyMonitorSerialConnector.java

```diff
@@ -91,23 +102,56 @@ public void disconnect() throws OpenEnergyMonitorException {
             connect();
         }
 
-        throw new OpenEnergyMonitorException("Not implemented");
-
-        /*
-         * byte[] buffer = new byte[1024];
-         * 
-         * int len = -1;
-         * try {
-         * while ((len = this.in.read(buffer)) > 0) {
-         * for (int i = 0; i < len; i++) {
-         * 
-         * }
-         * }
-         * } catch (IOException e) {
-         * throw new OpenEnergyMonitorException(
-         * "Error occured while receiving data", e);
-         * }
-         */
+        byte[] buffer = new byte[100];
+        StringBuilder sb = new StringBuilder();
 
+        int len = -1;
+        try {
+            while ((len = this.in.read(buffer)) > 0) {
+                for (int i = 0; i < len; i++) {
+                    logger.trace("Received char: {}", buffer[i]);
+                    if (buffer[i] != 0x0A) {
+                        if (buffer[i] != 0x0D) {
+                            sb.append((char) buffer[i]);
+                        }
+                    } else {
+                        return parseLine(sb.toString());
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new OpenEnergyMonitorException("Error occured while receiving data", e);
+        }
+
+        return null;
+    }
+
+    private byte[] parseLine(String line) throws OpenEnergyMonitorException {
+        logger.trace("Received line: {}", line);
+        String[] bytes = line.toString().split(" ");
+        ByteBuffer bytebuf = ByteBuffer.allocate(bytes.length);
+
+        try {
+            for (int j = 0; j < bytes.length; j++) {
+                if (!bytes[j].isEmpty()) {
+                    byte b = (byte) Integer.parseInt(bytes[j]);
+                    bytebuf.put(b);
+                }
+            }
+        } catch (NumberFormatException e) {
+            // Rubbish received from the serial port, ignore the whole frame
+            throw new OpenEnergyMonitorException("Error occured while receiving data", e);
+        }
+
+        return bytebuf.array();
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent arg0) {
+        try {
+            logger.trace("RXTX library CPU load workaround, sleep forever");
+            Thread.sleep(Long.MAX_VALUE);
+        } catch (InterruptedException e) {
+        }
```

Comment:
Should at least have a DEBUG statement in this catch.

Reply:
Well, InterruptedException will occur always when binding is shutdown, so logging is pretty useless. Same code without debug is used in most of the bindings which have some kind of serial port support.

Reply:
It's poor form to fail to log a handled exception.

If this argument is unpersuasive, I'll accept your choice.